pdr-api/layers/dataUtils/siteUtils.js

```diff
@@ -27,6 +56,280 @@ async function getSitesForProtectedArea(identifier) {
+      ':effectiveDate': { S: body.effectiveDate }
+    };
+    let updateExpression = ['SET updateDate = :updateDate, #status = :status, lastModifiedBy = :lastModifiedBy, effectiveDate = :effectiveDate'];
+
+    // Add optional fields to update
+    if (updateType !== REPEAL_UPDATE_TYPE) {
+      for (const field of OPTIONAL_PUT_FIELDS) {
+        if (body.hasOwnProperty(field)) {
+          updatedAttributeValues[`:${field}`] = { S: body[field] || '' };
+          updateExpression.push(`${field} = :${field}`);
+        }
+      }
+    }
+
+    // Build update transaction for site object
+    let siteUpdate = {
+      TableName: TABLE_NAME,
+      Key: {
+        pk: { S: identifier },
+        sk: { S: 'Details' }
+      },
+      ExpressionAttributeValues: updatedAttributeValues,
+      ExpressionAttributeNames: { '#status': 'status' },
+      UpdateExpression: updateExpression.join(', '),
+      ConditionExpression: 'updateDate = :lastVersionDate',
+    };
+
+    // If major change (not repealed), deny if legalName is the same
+    if (updateType === MAJOR_UPDATE_TYPE) {
+      siteUpdate.ExpressionAttributeValues[':oldLegalName'] = { S: body.legalName };
+      siteUpdate.ConditionExpression += ' AND legalName <> :oldLegalName';
+    }
+
+    // Add to transaction list
+    let putTransactions = [{ action: 'Update', data: siteUpdate }];
+
+    logger.debug('Site record update parameters:', siteUpdate);
+
+    // 2. Update the protected area site object (pk: <pAreaId>, sk: Site::<siteId>)
+    // Only update if the displayName property is changing
+    if (body?.displayName) {
+      const paSitePut = createPASiteUpdate(identifier, body.displayName);
+      putTransactions.push({ action: 'Update', data: paSitePut });
+    }
+
+    // 3. Create changelog item
+    // Only create a changelog item if doing a major change or repeal
+    if (updateType === MAJOR_UPDATE_TYPE || updateType === REPEAL_UPDATE_TYPE) {
+      putTransactions.push(await createChangeLogPut(identifier, body, user, currentTimeISO, newStatus));
+    }
+
+    // Return an array of put actions
+    return putTransactions;
+
+  } catch (error) {
+    logger.error('Failed to create site update transaction');
+    throw error;
+  }
+};
+
+/**
+ * Creates a transaction to update the display name of a protected area site.
+ * @function createPASiteUpdate
+ * @param {string} identifier - The identifier of the site.
+ * @param {string} displayName - The new display name for the site.
+ * @returns {Object} - The update transaction for the protected area site object.
+ * @throws {Exception} Throws an exception if the transaction creation fails.
+ */
+const createPASiteUpdate = function (identifier, displayName) {
+  try {
+
+    // Build update transaction for protected area site object
+    const paSiteUpdate = {
+      TableName: TABLE_NAME,
+      Key: {
+        pk: { S: identifier.split('::')[0] },
+        sk: { S: `Site::${identifier.split('::')[2]}` }
+      },
```

Comment:
Add some error handling in case the size of the split is not as expected before dereferencing the split like this